extension/lib/respository/BigcommerceCustomerRepository.js

```diff
@@ -0,0 +1,29 @@
+class BigcommerceCustomerRepository {
+  constructor (apiClientV2) {
+    this.apiClientV2 = apiClientV2
+  }
+
+  static getInstance (apiClientV2) {
```

Comment:
Isn't getInstance supposed to be used for singletons? I'd prefer factory method like `create`.

Reply:
Right, done.